### PR TITLE
Teacher Override - Song Filter

### PR DESCRIPTION
**Pull request by epeach**
Part of Age-Filtering

Builds on: https://github.com/code-dot-org/code-dot-org/pull/25741

Appending `songfilter=on` to the URL will override filtering songs based on age. Persists as long as the url contains `songfilter=on`.

Includes tests.

### CODE REVIEW

**Comment by islemaster** [comment]
Are we sure this is the right approach? Using the querystring to override song filtering seems too easy to bypass.  How did we decide on this approach to this feature?

**Comment by epeach** [comment]
I think we do want it to be very easy to bypass so that teachers who would prefer all their students have access to only filtered songs can provide this link to their student. Spec here: https://docs.google.com/document/d/1GXo6xekcqjIQ-UGjxcGhfKxZexxxYofck8C_MLUppVM/edit#

@ryansloan - can you provide more context of how we arrived at this solution?

**Comment by ryansloan** [comment]
@islemaster Yep, that's right - it's by design that it's easy to bypass. We want to give teachers who want to be in control of the filtering the ability to just write a different URL on the board

**Comment by islemaster** [comment]
Having `songFilter=off` as a "hidden" feature to bypass song filtering makes sense to me.  Having `songFilter=on` to enforce song filtering seems nearly useless, given how easily a student could discover, omit and/or change this setting once used, _and_ given most teachers will want to use our shortcode `<pegasus>/danceparty` which doesn't pass queryparams through the redirect.

**Comment by islemaster** [comment]
Maybe that's an important question: Is there hidden work here?  Do we need to pass queryparams through that shortcode and/or from level to level in the script?

**Comment by epeach** [comment]
I see what you mean about how easy it is to bypass the filter. @ryansloan Is this something we could allow teachers to turn on/off for their entire sections from the teacher dashboard?

**Comment by ryansloan** [comment]
Ok, I talked with LT about this. We are going to only let you override to turn the filter on. 
`songfilter=off` should be a no-op.

**Comment by epeach** [comment]
@ryansloan So just to confirm, once you turn the filter on for yourself, there is no way to turn it off?

**Comment by ryansloan** [comment]
@epeach and I chatted about this. We are not trying to build a robust filtering system, only something that teachers can opt to use if they want to in their classroom. They can enforce the content restrictions using the same classroom management practices they use for computer time in general to ensure students are using the URL. It is trivial to get around, and that's ok.

We don't want any browser/user to be "forever filtered". Can we expire the cookie with the session?

**Comment by epeach** [comment]
@islemaster PTAL, thanks!

**Comment by islemaster** [comment]
I assume you didn't mean to check in a `schema_cache.dump`?

File: node_modules/.yarn-integrity [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (islemaster) [inline node_modules/.yarn-integrity]: I also have no idea what this file is, but it's probably specific to your machine.

File: node_modules/@code-dot-org/dance-party [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../.nvm/versions/node/v6.9.0/lib/node_modules/@code-dot-org/dance-party
```
Comment (islemaster) [inline node_modules/@code-dot-org/dance-party]: And I'd remove this symlink too.

**Pull request merged**